tensorflow_model_optimization/g3doc/guide/experimental/combine/collaborative_optimization.md

```diff
@@ -0,0 +1,107 @@
+* [Cluster preserving quantization aware training](https://github.com/tensorflow/model-optimization/pull/617) (CQAT)
+* Sparsity and cluster preserving quantization aware training (PCQAT) - WIP
+
+These provide several deployment paths that could be used to compress a machine learning model and to take advantage of hardware acceleration at inference time. The diagram below demonstrates several deployment paths that can be explored in search for the model with desired deployment characteristics, where the leaf nodes are deployment-ready models, meaning they are partially or fully quantized and in tflite format. The green fill indicates steps where retraining/fine-tuning is required and a dashed red border highlights the collaborative optimization steps. The technique used to obtain a model at a given node is indicated in the corresponding label.
+
+![collaborative optimization](images/collaborative_optimization.png "collaborative optimization")
+
+The direct, quantization-only (post-training or QAT) deployment path is omitted in the figure above.
+
+The idea is to reach the fully optimized model at the third level of the above deployment tree; however, any of the other levels of optimization could prove satisfactory and achieve the required inference latency/accuracy trade-off, in which case no further optimization is needed. The recommended training process would be to iteratively go through the levels of the deployment tree applicable to the target deployment scenario and see if the model fulfils the inference latency requirements and, if not, use the corresponding collaborative optimization technique to compress the model further and repeat until the model is fully optimized (pruned, clustered, and quantized), if needed.
+
+The figure below shows the density plots of sample weight kernel going through the collaborative optimization pipeline.
+
+![collaborative optimization density plot](images/collaborative_optimization_dist.png "collaborative optimization density plot")
+
+The result is a quantized deployment model with a reduced number of unique values as well as a significant number of sparse weights, depending on the target sparsity specified at training time. Other than the significant model compression advantages, specific hardware support can take advantage of these sparse, clustered models to significantly reduce inference latency.
+
+
+## Results
+
+Below are some accuracy and compression results we obtained when experimenting with PQAT and CQAT collaborative optimization paths.
+
+### Sparsity-preserving Quantization aware training (PQAT)
+
+<style type="text/css">
+	table.tableizer-table {
+		font-size: 12px;
+		border: 1px solid #CCC;
+		font-family: Arial, Helvetica, sans-serif;
+	}
+	.tableizer-table td {
+		padding: 4px;
+		margin: 3px;
+		border: 1px solid #CCC;
+	}
+	.tableizer-table th {
+		background-color: #104E8B;
+		color: #FFF;
+		font-weight: bold;
+	}
+</style>
```

Comment:
Not sure if it is just my version but when I view this file, I still see these style tags.

Reply:
Yes, just saw that it's rendered differently in github markdown than it does locally for me! There's an issue with inline  embedding of CSS style, so I reformatted the table without it and pushednew commits. Should be ok now!